src/app/services/movies/movies.service.ts

```diff
@@ -0,0 +1,52 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from "@angular/common/http";
+import {MoviesDto} from "../../model/movies-dto";
+import {Observable} from "rxjs";
+import {Movie} from "../../model/movie/movie";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MoviesService {
```

Comment:
the service is a little bit messy, it's a bad mixture of imperative with reactive approach. of course it's better to use good reactive design if you use rxjs, but it's okay to move forward and refactor it later.

Reactive approach would meaning defining and composing data streams with rxjs pipes, defining action subjects which will start a chain of reaction for your defined data-sources. There are just two links to start digging into rxjs approach, they don't contain a direct answer for your problem but it's for sure a good read. https://blog.angular-university.io/how-to-build-angular2-apps-using-rxjs-observable-data-services-pitfalls-to-avoid/ https://blog.openreplay.com/functional-reactive-programming-in-angular-with-rxjs check it out, you can try using more of it in your future work